Userland/Libraries/LibCore/System.cpp

```diff
@@ -178,6 +179,64 @@ ErrorOr<void> profiling_free_buffer(pid_t pid)
     int rc = ::profiling_free_buffer(pid);
     HANDLE_SYSCALL_RETURN_VALUE("profiling_free_buffer", rc, {});
 }
+
+ErrorOr<void> exec(StringView filename, Span<StringView> arguments, SearchInPATH search_in_path, Optional<Span<StringView>> environment)
+{
+    auto copy_strings = [](Span<StringView> const& source, Syscall::StringListArgument& dest) {
+        dest.strings = new Syscall::StringArgument[source.size()];
+        dest.length = source.size();
+        for (size_t i = 0; i < source.size(); ++i) {
+            dest.strings[i] = { source[i].characters_without_null_termination(), source[i].length() };
+        }
+    };
+
+    Syscall::SC_execve_params params;
+    copy_strings(arguments, params.arguments);
+
+    if (environment.has_value()) {
+        copy_strings(environment.value(), params.environment);
+    } else {
+        size_t env_count = 0;
+        for (size_t i = 0; environ[i]; ++i)
+            ++env_count;
+        params.environment.strings = new Syscall::StringArgument[env_count];
```

Comment:
This feels kinda weird. Can we convert a Span<StringView> into a StringListArgument without allocating? Their representation should be identical in memory. Not that it matters when we're about to do a bunch of allocations in the kernel when we exec() for the new process and such, but technically both of these string list arguments are leaked.

Reply:
I ended up replacing the implementation to call the libc functions, so this no longer applies.